### DISCUSSION TITLE
Suggestion: reword bootstrap resolvers section

### DISCUSSION THREAD

**Discussion by slopsommelier** [DNSCrypt/dnscrypt-proxy#2670] [Ideas]
Based on my interactions with the developer of inviZible, it seems that many do not understand that the bootstrap resolvers are not necessary if you use a stamp that includes the IP address for your server.

By removing/commenting out the bootstrap resolvers line (or setting them to loopback, null, 0.0.0.0, etc.) you can ensure that, even if for some reason you are unable to connect to your chosen resolver, the proxy will fail to connect instead of sending a request on port 53.

While this is a minor improvement in privacy/security, it's still worth mentioning, since on unrooted Android devices for example you cannot block port 53 (especially problematic for WWAN).

So my suggestion is to clarify with something like
'you can comment out this line or set these to an invalid address (e.g. 127.0.0.1:53) if you use a DNS stamp that already contains an IP address'

**Comment by lifenjoiner** [comment]
https://github.com/DNSCrypt/dnscrypt-proxy/blob/de16f22c0cea8606d0a270481633824f175e38a3/dnscrypt-proxy/example-dnscrypt-proxy.toml#L241-L256
Read it carefully. Seems you don't get it. Bootstrap resolvers may not be used as you thought.

> removing/commenting out the bootstrap resolvers line

1. For most people, who use the default sources (the initial resolvers list), it is required, when the app is launched the first time. Source file server names need to be resolved before downloading the files.
2. Only those who use a **static** stamp with IP server, can do it without failure. But, in that case, `bootstrap_resolvers` wil not be used at all.

> the proxy will fail to connect instead of sending a request on port 53

**No user queries will ever be leaked through these resolvers**.
Many people ignore this, but thought it as them as fallback. But, NO! Explained many times. They are not used to resolve user queries at all.

**Reply by slopsommelier** [reply]
I do understand it

Which is why I am saying, you should clarify that if you use a stamp (like I do) you can delete them/comment out this line. They are not needed. 

You can write explicitly:
If your stamp includes an IP address, you may delete these resolvers, set them to an invalid address, or comment out this line. 

Then you can block port 53 in your system firewall/router whatever. 

This is exactly what I do, but many people think you cannot do it because it does not explicitly say you can remove them.

**Reply by lifenjoiner** [reply]
If not be used, no difference to delete or not, but causing general users confused and then make mistakes.
